apps/web/src/pages/TalentNominations/NominationGroup/components/FullCareerExperiences.tsx

```diff
@@ -0,0 +1,251 @@
+import { useState } from "react";
+import { useIntl } from "react-intl";
+import NewspaperIcon from "@heroicons/react/24/outline/NewspaperIcon";
+
+import { Button, Heading, Well } from "@gc-digital-talent/ui";
+import { AwardExperience, Experience } from "@gc-digital-talent/graphql";
+
+import experienceMessages from "~/messages/experienceMessages";
+import useControlledCollapsibleGroup from "~/hooks/useControlledCollapsibleGroup";
+import {
+  compareByDate,
+  isAwardExperience,
+  isCommunityExperience,
+  isEducationExperience,
+  isPersonalExperience,
+  isWorkExperience,
+} from "~/utils/experienceUtils";
+
+import ExperienceByTypeAccordion from "./ExperienceByTypeAccordion";
+
+interface FullCareerExperiencesProps {
+  experiences?: Omit<Experience, "user">[];
+  shareProfile?: boolean;
+}
+
+const FullCareerExperiences = ({
+  experiences,
+  shareProfile,
+}: FullCareerExperiencesProps) => {
+  const intl = useIntl();
+  const [selectedView, setSelectedView] = useState<"type" | "workStream">(
+    "type",
+  );
+  const awardExperiences =
+    experiences
+      ?.filter(isAwardExperience)
+      .map(
+        (award: Omit<AwardExperience, "user">) =>
+          ({
+            ...award,
+            startDate: award.awardedDate,
+            endDate: award.awardedDate,
+          }) as AwardExperience & { startDate: string; endDate: string },
+      )
+      .sort(compareByDate) ?? [];
+  const communityExperiences =
+    experiences?.filter(isCommunityExperience).sort(compareByDate) ?? [];
+  const educationExperiences =
+    experiences?.filter(isEducationExperience).sort(compareByDate) ?? [];
+  const personalExperiences =
+    experiences?.filter(isPersonalExperience).sort(compareByDate) ?? [];
+  const workExperiences =
+    experiences?.filter(isWorkExperience).sort(compareByDate) ?? [];
+
+  const experienceSections = [
```

Comment:
Another bad smell related to the collapse/expand setup is that you need to calculate all the "by type" section IDs outside of the actual "by type" component.  There might be a better way to organize this?

Reply:
ok ..changed to Yoni's pattern for the profile page here [9464640](https://github.com/GCTC-NTGC/gc-digital-talent/pull/13341/commits/9464640af2fd961dca28e6e5db916097be7426d8)

Reply:
Nice, this looks much better!  :+1: